### PR TITLE
Example creating dataset with derived array

### PR DESCRIPTION
**Pull request by jjdelc**

### CODE REVIEW

File: source/includes/_feature-importing.md [hunk 1]
```diff
@@ -128,6 +128,139 @@ object is explicitly unordered. If you wish the variables to have an order,
...
+{
+    "element": "shoji:entity",
+    "body": {
+      "name": "Dataset with derived arrays",
+      "table": {
+        "metadata": {
+          "CA3": {
+            "name": "cat array 3", 
+            "derivation": {
+              "function": "array",
```
Comment (nealrichardson) [inline source/includes/_feature-importing.md:208]: Is the "array" function documented elsewhere? If not, please add it to whatever section where you did document all of the "combine" functions for the scrunch project.

Please link over to that section in the discussion before this example.

Make sure that in the "array" function docs that it is clear that the "references" members of the "map" are optional.

Comment (jjdelc) [reply source/includes/_feature-importing.md:208]: The references aren't optional. Else you get a virtual variable without name/alias and shit breaks.

Comment (nealrichardson) [reply source/includes/_feature-importing.md:208]: Then how does this work? 

https://github.com/Crunch-io/rcrunch/blob/master/R/make-array.R#L79-L94
https://github.com/Crunch-io/rcrunch/blob/master/tests/testthat/test-derive-array.R

Comment (nealrichardson) [reply source/includes/_feature-importing.md:208]: It appears that the derived arrays take the references of the subvariables they derive from.

Comment (jjdelc) [reply source/includes/_feature-importing.md:208]: will check

Comment (jjdelc) [reply source/includes/_feature-importing.md:208]: Added docs
https://github.com/Crunch-io/apidocs/commit/1e2cabe7fad13f234c0d0261d881e239ccbd7133

**Pull request merged**